openapi_python_client/parser/properties/union.py

```diff
@@ -80,22 +80,44 @@ def build(
         sub_properties: list[PropertyProtocol] = []
 
         type_list_data = []
-        if isinstance(data.type, list):
+        if isinstance(data.type, list) and not (data.anyOf or data.oneOf):
             for _type in data.type:
                 type_list_data.append(data.model_copy(update={"type": _type, "default": None}))
 
-        for i, sub_prop_data in enumerate(chain(data.anyOf, data.oneOf, type_list_data)):
-            sub_prop, schemas = property_from_data(
-                name=f"{name}_type_{i}",
-                required=True,
-                data=sub_prop_data,
-                schemas=schemas,
-                parent_name=parent_name,
-                config=config,
-            )
-            if isinstance(sub_prop, PropertyError):
-                return PropertyError(detail=f"Invalid property in union {name}", data=sub_prop_data), schemas
-            sub_properties.append(sub_prop)
```

Comment:
Here's the main fix. Basically, when we were iterating through the variants of a union, we were making up inline type names like "MyProperty_type_0", "MyProperty_type_1", etc... which would be used in case the variant was an inline schema for an object or an enum. However, in cases where _only one_ of the variants is a schema like that— and the others are basic types like int or null, which don't require their own class name— this was resulting in us making up a new type when none was needed.